### ISSUE TITLE
Replace initialization "hack' with a proper event listener

### ISSUE THREAD

**Issue by markcarver** [truckingsim/Ajax-Bootstrap-Select#48]
https://github.com/truckingsim/Ajax-Bootstrap-Select/blob/master/src/js/classes/AjaxBootstrapSelect.js#L246-L251

This code uses a `setTimeout` in a very hack-ish way (basically, hope that bootstrap-select has been initialized first within 500ms). This is really unreliable and should be listening for an event instead.

We're actually waiting for the bootstrap-select project to create an `initialized` event that they themselves will trigger after their plugin has been created (see https://github.com/silviomoreto/bootstrap-select/issues/699).

This will allow us to ensure that bootstrap-select has indeed initialized before attempting to initialize our plugin.

**Comment by truckingsim** [comment]
Looks like this has been added and will be part of 1.7.0.

**Comment by markcarver** [comment]
Indeed, as `loaded.bs.select`. Also, makes me wonder if we should be doing any sort of "version check"?

**Comment by truckingsim** [comment]
I think we'll have to on this update as it won't work without the events.
